### DISCUSSION TITLE
How to manage MIDL configuration projet with lua ?

### DISCUSSION THREAD

**Discussion by JohPac** [premake/premake-core#1984] [Q&A]
Hi all,

I'm currently migrating projects Visual with ocx (MIDL) under premake.

In lua, when I add file .odl in files section, so **ItemGroup/Midl** is well add in vcxproj 👍 
```
  <ItemGroup>
    <Midl Include="..\test.odl">
    </Midl>
  </ItemGroup>
```

But to compile it's missing in vcxproj : **ItemDefinitionGroup/Midl** to include TypeLibraryName test.tlb

for exemple, in original vcxproj (no generate with premake)
```
<ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
<Midl>
      <PreprocessorDefinitions>NDEBUG;%(PreprocessorDefinitions)</PreprocessorDefinitions>
      <MkTypLibCompatible>false</MkTypLibCompatible>
      <SuppressStartupBanner>true</SuppressStartupBanner>
      <TargetEnvironment>Win32</TargetEnvironment>
      <TypeLibraryName>$(ProjectDir)$(Configuration)\test.tlb</TypeLibraryName>
    </Midl>
</ItemDefinitionGroup>
```

I don't find in docs/Stack Overflow any info how to add this in lua configuration.

Can you confirme, that is it possible in the current version of premake 5.0b2 ?

Thx,

**Comment by samsinsane** [comment]
We only support excluding a `.idl` file from the build, you can see this here: https://github.com/premake/premake-core/blob/ff5328c214d1f91e3eeb51bae1f8be9133f67ea8/modules/vstudio/vs2010_vcxproj.lua#L959-L980

PRs are welcome for expanding upon this, and if you're looking for a temporary workaround, you can use the override system with something like this:
```lua
require "vstudio"

local function someFileElement(cfg, file)
  premake.vstudio.vc2010.element("someFileElement", nil, "value")
end

local function someFileCfgElement(filecfg, condition)
  premake.vstudio.vc2010.element("someFileCfgElement", condition, "value")
end

premake.override(premake.vstudio.vc2010, "emitFiles", function (oldfn, prj, group, tag, fileFunc, fileCfgFunc, checkFunc)
  if tag == "Midl" then
    fileFunc = table.join(fileFunc or {}, {
      someFileElement,
    }

    fileCfgFunc = table.join(fileCfgFunc or {}, {
      someFileCfgElement,
    }
  end

  return oldfn(prj, group, tag, fileFunc, fileCfgFunc, checkFunc)
end)
```

`someFileElement` will emit an element for each file (like `SupressStartUpBanner` if you want it to always be `true`), while `someFileCfgElement` will emit an element for each file in each configuration (like `TargetEnvironment` if you're building for both `Win32` and `x64`).

**Reply by JohPac** [reply]
Thank you for your advice, we try this ASAP
